io/plugins/eu.esdihumboldt.hale.io.xls/src/eu/esdihumboldt/hale/io/xls/AbstractAnalyseTable.java

```diff
@@ -136,17 +139,21 @@ protected void analyseHeader(Sheet sheet) {
 	protected abstract void headerCell(int num, String text);
 
 	/**
-	 * Analyse the table content.
+	 * Analyse the table content if skipNlines <=0 that don't analyse first row,
+	 * which has already been analyse into the header else analyse starting with
+	 * the skip line
 	 * 
 	 * @param sheet the table sheet
+	 * @param skipNlines number of lines to skip
 	 */
-	private void analyseContent(Sheet sheet) {
-		// for each row starting from the second
-		for (int i = 1; i <= sheet.getLastRowNum(); i++) {
+	private void analyseContent(Sheet sheet, int skipNlines) {
+		int startLine = skipNlines;
+		if (skipNlines <= 0) {
```

Comment:
Because the first line has already been analyse in the header if skipNlines <= 0.
And if the header is a header or just a row with data the first line will be added to the instances anyhow.

Reply:
If that's the argument then in my view `analyseHeader` should not only be called when `skipNLines` is zero or less (see other comment).

Reply:
> the first line will be added to the instances anyhow

I think I found the [code](https://github.com/halestudio/hale/blob/3de05597fc72a1b6c7b0639e2b6bba50374088e6/io/plugins/eu.esdihumboldt.hale.io.xls/src/eu/esdihumboldt/hale/io/xls/reader/XLSInstanceReader.java#L176) you are referring to.

But wouldn't it make more sense if you introduce to the API in `AbstractAnalyseTable` that lines should be skipped, implying that the skipped lines are not part of the data, that in case there are no line to be skipped and thus no dedicated header, that it would report all rows as data rows?
In my view the instance reader should just not add the header then. This was done previously because `AbstractAnalyseTable` so far represented the concept of a table where the first row is the header and the rest are data.

Reply:
Also in my view the header should not be added.
Doesn't make any sense to have the header as a data as long as doesn't contain any data.

Reply:
So do we then agree that the following is true? 

- the instance reader should not add the header provided by `AbstractAnalyseTable` as instance
- `AbstractAnalyseTable` should call `analyseRow` for the first row if `skipNLines` is less or equal to zero

Reply:
@stempler yes. and the code at the moment is exactly in line with what you have mention.

Reply:
> and the code at the moment is exactly in line with what you have mention.

I'm afraid we disagree there.

- My first point would mean removing the whole section on `if (skipNlines <= 0)` from the instance reader (see new comment).
- My second point would mean not setting `startLine` to `1` here, so the data row is part of the rows that are analysed with `analyseRow`